src/main/java/io/quarkus/gizmo2/impl/EqualsHashCodeToStringGenerator.java

```diff
@@ -0,0 +1,297 @@
+        MethodDesc objectArrayEquals = MethodDesc.of(Arrays.class, "equals", boolean.class, Object[].class, Object[].class);
+        MethodDesc arrayDeepEquals = MethodDesc.of(Arrays.class, "deepEquals", boolean.class, Object[].class, Object[].class);
+
+        ClassDesc thisClass = cc.type();
+
+        cc.method("equals", mc -> {
+            mc.public_();
+            mc.returning(boolean.class);
+            ParamVar other = mc.parameter("other", Object.class);
+            mc.body(b0 -> {
+                b0.if_(b0.eq(mc.this_(), other), b1 -> {
+                    b1.return_(true);
+                });
+                b0.ifNotInstanceOf(other, thisClass, b1 -> {
+                    b1.return_(false);
+                });
+
+                Expr otherCast = b0.define("other", b0.cast(other, thisClass));
+                for (FieldDesc field : fields) {
+                    if (!field.owner().equals(thisClass)) {
+                        throw new IllegalArgumentException("Field does not belong to " + thisClass.displayName() + ": " + field);
+                    }
+
+                    Expr thisValue = b0.define("thisValue", b0.get(mc.this_().field(field)));
+                    Expr thatValue = b0.define("thatValue", b0.get(otherCast.field(field)));
+                    switch (field.type().descriptorString()) {
+                        // boolean, byte, short, int, long, char
+                        case "Z", "B", "S", "I", "J", "C" ->
+                            b0.if_(b0.ne(thisValue, thatValue), b1 -> {
+                                b1.return_(false);
+                            });
+                        // float
+                        case "F" -> {
+                            // this is consistent with Arrays.equals() and it's also what IntelliJ generates
+                            Expr thisBits = b0.invokeStatic(floatToIntBits, thisValue);
+                            Expr thatBits = b0.invokeStatic(floatToIntBits, thatValue);
+                            b0.if_(b0.ne(thisBits, thatBits), b1 -> {
+                                b1.return_(false);
+                            });
+                        }
+                        // double
+                        case "D" -> {
+                            // this is consistent with Arrays.equals() and it's also what IntelliJ generates
+                            Expr thisBits = b0.invokeStatic(doubleToLongBits, thisValue);
+                            Expr thatBits = b0.invokeStatic(doubleToLongBits, thatValue);
+                            b0.if_(b0.ne(thisBits, thatBits), b1 -> {
+                                b1.return_(false);
+                            });
+                        }
+                        // boolean[]
+                        case "[Z" ->
+                            b0.unless(b0.invokeStatic(booleanArrayEquals, thisValue, thatValue), b1 -> {
```

Comment:
Since the method descriptors are already being acquired within the above method, it might be simpler overall to combine similar ccases and have it be like: `b0.invokeStatic(ClassMethodDesc.of(CD_Arrays, "equals", field.type(), field.type())` without the declarations above. WDYT?

Reply:
That didn't occur to me at all, makes sense. Will do.

Reply:
Done.